pkg/retry/policy.go

```diff
@@ -166,6 +165,8 @@ const (
 type ShouldResultRetry struct {
 	ErrorRetry func(err error, ri rpcinfo.RPCInfo) bool
 	RespRetry  func(resp interface{}, ri rpcinfo.RPCInfo) bool
+	// the requests that are not idempotent cannot do timeout retry
```

Comment:
disable the default timeout retry in specific scenarios (e.g. the requests are not non-idempotent)

Reply:
done